src/Grpc.AspNetCore.Server/Internal/CallHandlers/ClientStreamingServerCallHandler.cs

```diff
@@ -72,6 +73,13 @@ internal class ClientStreamingServerCallHandler<TService, TRequest, TResponse> :
 
         protected override async Task HandleCallAsyncCore(HttpContext httpContext)
         {
+            // Disable request body data rate for client streaming
+            var minRequestBodyDataRateFeature = httpContext.Features.Get<IHttpMinRequestBodyDataRateFeature>();
+            if (minRequestBodyDataRateFeature != null)
```

Comment:
Technically, since we are running on Kestrel, we are guaranteed that this isn't null. Could switch this to a Debug.Assert and update the functional tests to add a dummy IHttpMinRequestBodyDataRateFeature. Currently this is probably overly forgiving.

Reply:
I think this is fine.

Move setting it to null to a method and reuse from duplex and client streaming handlers.